DeprecationWarning: msort is deprecated

I feel like the universe doesn't want this feature freeze to happen...

```
_________________________ TestSortFunctions.test_msort _________________________

self = <astropy.units.tests.test_quantity_non_ufuncs.TestSortFunctions object at 0x7f087f2cd790>

    def test_msort(self):
>       self.check(np.msort)

.../astropy/units/tests/test_quantity_non_ufuncs.py:1508: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.../astropy/units/tests/test_quantity_non_ufuncs.py:66: in check
    o = func(self.q, *args, **kwargs)
<__array_function__ internals>:200: in msort
    ???
.../astropy/units/quantity.py:1736: in __array_function__
    return super().__array_function__(function, types, args, kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

a = <Quantity [[0.  , 0.25, 0.5 ],
           [0.75, 1.  , 1.25],
           [1.5 , 1.75, 2.  ]] m>

    @array_function_dispatch(_msort_dispatcher)
    def msort(a):
...
        # 2022-10-20 1.24
>       warnings.warn(
            "msort is deprecated, use np.sort(a, axis=0) instead",
            DeprecationWarning,
            stacklevel=3,
        )
E       DeprecationWarning: msort is deprecated, use np.sort(a, axis=0) instead

.../numpy/lib/function_base.py:3682: DeprecationWarning
```

--- a/astropy/units/tests/test_quantity_non_ufuncs.py
+++ b/astropy/units/tests/test_quantity_non_ufuncs.py
@@ -11,7 +11,7 @@ from astropy import units as u
 from astropy.units.quantity_helper.function_helpers import (
     ARRAY_FUNCTION_ENABLED, DISPATCHED_FUNCTIONS, FUNCTION_HELPERS, IGNORED_FUNCTIONS,
     SUBCLASS_SAFE_FUNCTIONS, TBD_FUNCTIONS, UNSUPPORTED_FUNCTIONS)
-from astropy.utils.compat import NUMPY_LT_1_23
+from astropy.utils.compat import NUMPY_LT_1_23, NUMPY_LT_1_24
 
 needs_array_function = pytest.mark.xfail(
     not ARRAY_FUNCTION_ENABLED,
@@ -1500,12 +1500,17 @@ class TestSortFunctions(InvariantUnitTestSetup):
     def test_sort(self):
         self.check(np.sort)
 
+    def test_sort_axis(self):
+        self.check(np.sort, axis=0)
+
+    @pytest.mark.skipif(not NUMPY_LT_1_24,
+                        reason='np.msort is deprecated')
+    def test_msort(self):
+        self.check(np.msort)
+
     @needs_array_function
     def test_sort_complex(self):
         self.check(np.sort_complex)
-
-    def test_msort(self):
-        self.check(np.msort)
 
     def test_partition(self):
         self.check(np.partition, 2)

--- a/astropy/utils/masked/tests/test_function_helpers.py
+++ b/astropy/utils/masked/tests/test_function_helpers.py
@@ -19,7 +19,7 @@ import pytest
 from numpy.testing import assert_array_equal
 
 from astropy.units.tests.test_quantity_non_ufuncs import get_wrapped_functions
-from astropy.utils.compat import NUMPY_LT_1_23
+from astropy.utils.compat import NUMPY_LT_1_23, NUMPY_LT_1_24
 from astropy.utils.masked import Masked, MaskedNDArray
 from astropy.utils.masked.function_helpers import (
     APPLY_TO_BOTH_FUNCTIONS, DISPATCHED_FUNCTIONS, IGNORED_FUNCTIONS, MASKED_SAFE_FUNCTIONS,
@@ -1072,6 +1072,8 @@ class TestSortFunctions(MaskedArraySetup):
         expected = ma[indx]
         assert_masked_equal(o, expected)
 
+    @pytest.mark.skipif(not NUMPY_LT_1_24,
+                        reason='np.msort is deprecated')
     def test_msort(self):
         o = np.msort(self.ma)
         expected = np.sort(self.ma, axis=0)

